source/core/module_typeHelpers.js

```diff
@@ -0,0 +1,340 @@
+            return Module._TypeRef_IsEnum(typeRef) !== 0;
+        };
+
+        Module.typeHelpers.isFloat = function (typeRef) {
+            return Module._TypeRef_IsFloat(typeRef) !== 0;
+        };
+
+        Module.typeHelpers.isString = function (typeRef) {
+            return Module._TypeRef_IsString(typeRef) !== 0;
+        };
+
+        Module.typeHelpers.isPath = function (typeRef) {
+            return Module._TypeRef_IsPath(typeRef) !== 0;
+        };
+
+        Module.typeHelpers.isTimestamp = function (typeRef) {
+            return Module._TypeRef_IsTimestamp(typeRef) !== 0;
+        };
+
+        Module.typeHelpers.isComplex = function (typeRef) {
+            return Module._TypeRef_IsComplex(typeRef) !== 0;
+        };
+
+        Module.typeHelpers.isAnalogWaveform = function (typeRef) {
+            return Module._TypeRef_IsAnalogWaveform(typeRef) !== 0;
+        };
+
+        var typeHandlers = [
+            {
+                typeChecker: Module.typeHelpers.isBoolean,
+                dispatcher: dispatchVisitBoolean
+            },
+            {
+                // Enum is wrapping an integer, so it needs to be evaluated first.
+                typeChecker: Module.typeHelpers.isEnum,
+                dispatcher: dispatchVisitEnum
+            },
+            {
+                typeChecker: Module.typeHelpers.isInteger,
+                dispatcher: dispatchVisitInteger
+            },
+            {
+                typeChecker: Module.typeHelpers.isFloat,
+                dispatcher: dispatchVisitFloat
+            },
+            {
+                // String is an array of UTF-8 chars so it is evaluated before array.
+                typeChecker: Module.typeHelpers.isString,
+                dispatcher: dispatchVisitString
+            },
+            {
+                typeChecker: Module.typeHelpers.isComplex,
+                dispatcher: dispatchVisitComplex
+            },
+            {
+                typeChecker: Module.typeHelpers.isAnalogWaveform,
+                dispatcher: dispatchVisitAnalogWaveform
+            },
+            {
+                typeChecker: Module.typeHelpers.isTimestamp,
+                dispatcher: dispatchVisitTimestamp
+            },
+            {
+                typeChecker: Module.typeHelpers.isPath,
+                dispatcher: dispatchVisitPath
+            },
+            {
+                typeChecker: Module.typeHelpers.isArray,
+                dispatcher: dispatchVisitArray
+            },
+            {
+                // Cluster is evaluated last because Complex, AnalogWaveform, Path and Timestamps
+                // are internally also clusters.
+                typeChecker: Module.typeHelpers.isCluster,
+                dispatcher: dispatchVisitCluster
+            }
+        ];
+
+        Module.typeHelpers.findTypeDispatcher = function (typeRef) {
+            var i = 0,
+                typeHandler;
+
+            for (i = 0; typeHandlers.length; i += 1) {
```

Comment:
Should be `i < typeHandlers.length`

Reply:
😮 you are right. dunno how that slip-through. I'll fix it later today.

Reply:
Seems like it couldn't have, it would be an infinite loop.

Reply:
Yeah, we are missing a test for a type that we cannot handle. I'll add that as well as fixing this.